### PR TITLE
1.2 auto publishing

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] Nodes can now be auto-published/unpublished on date 
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1845
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/634
https://github.com/open-orchestra/open-orchestra-model-interface/pull/217
https://github.com/open-orchestra/open-orchestra-provision/pull/40
https://github.com/open-orchestra/open-orchestra-docs/pull/230
https://github.com/open-orchestra/open-orchestra/pull/1028

### CODE REVIEW

**Comment by alavieille** [comment]
add OO-TAG

File: Backoffice/Manager/NodePublisher.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    ) {
+        $this->statusRepository = $statusRepository;
+        $this->nodeRepository = $nodeRepository;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @param ReadSiteInterface $site
+     *
+     * @return array | int   A published node list or an error code
```
Comment (alavieille) [inline Backoffice/Manager/NodePublisher.php]: remove space

File: Backoffice/Manager/NodePublisher.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        }
+        $this->objectManager->flush();
+
+       return $publishedNodes;
+    }
+
+    /**
+     * @param ReadSiteInterface $site
+     *
+     * @return array | int   An unpublished node list or an error code
```
Comment (alavieille) [inline Backoffice/Manager/NodePublisher.php]: remove space between type

File: Backoffice/Manager/NodePublisherInterface.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+Interface NodePublisherInterface
+{
+    const ERROR_NO_PUBLISH_FROM_STATUS = 1;
+    const ERROR_NO_PUBLISHED_STATUS = 2;
+    const ERROR_NO_UNPUBLISHED_STATUS = 3;
+
+    /**
+     * @param ReadSiteInterface $site
+     *
+     * @return array | int   A published node list or an error code
```
Comment (alavieille) [inline Backoffice/Manager/NodePublisherInterface.php]: remove space between type

File: BackofficeBundle/Command/OrchestraUnpublishNodeCommand.php [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * Execute command
+     * 
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return int 0 if everything went fine, or an error code
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (alavieille) [inline BackofficeBundle/Command/OrchestraUnpublishNodeCommand.php]: can you add a functional test to test this method ?

File: BackofficeBundle/Command/OrchestraPublishNodeCommand.php [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * Execute command
+     * 
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     *
+     * @return int 0 if everything went fine, or an error code
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (alavieille) [inline BackofficeBundle/Command/OrchestraPublishNodeCommand.php]: can you add a functional test to test this method

**Comment by ngilain** [comment]
@alavieille : update done + functionnal tests in https://github.com/open-orchestra/open-orchestra/pull/1028

File: Backoffice/Manager/NodePublisher.php [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+            $this->objectManager->persist($node);
+            $publishedNodes[] = array(
+                'BOLabel' => $node->getBoLabel(),
+                'version' => $node->getVersion(),
+                'language' => $node->getLanguage()
+            );
+        }
+        $this->objectManager->flush();
+
+       return $publishedNodes;
```
Comment (alavieille) [inline Backoffice/Manager/NodePublisher.php]: indent

**Comment by ngilain** [comment]
@alavieille : update done

**Comment by alavieille** [comment]
:+1:

**Pull request merged**